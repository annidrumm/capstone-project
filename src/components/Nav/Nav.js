import React from 'react'
import styled from 'styled-components/macro'
import GoBackArrow from '../../components/GoBackArrow/GoBackArrow'
import { ReactComponent as HeartIcon } from '../../icons/heart.svg'
import { Switch, Route, Link } from 'react-router-dom'

export default function NavHeader() {
  return (
    <>
      <GoBackArrow />
      <Switch>
        <Route>
          <Link to="/favourites">
            <ButtonStyled>
              <HeartIcon fill="var(--darkpink)" stroke="var(--darkpink)" />
            </ButtonStyled>
          </Link>
        </Route>
      </Switch>
    </>
  )
}

const ButtonStyled = styled.button`
  background: transparent;
  border-radius: 50%;
  text-align: center;
  height: 60px;
  width: 60px;
  display: block;
  margin: auto;
  margin-top: 15px;
  border: none;
  cursor: pointer;
`
