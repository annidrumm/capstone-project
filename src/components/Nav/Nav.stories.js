import React from 'react'
import Nav from './Nav.js'

export default {
  title: 'Components/Navigation',
  component: Nav,
}

export const navigation = () => {
  return <Nav />
}
