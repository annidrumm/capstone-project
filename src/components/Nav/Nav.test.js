import React from 'react'
import renderer from 'react-test-renderer'
import Nav from './Nav'
import '@testing-library/jest-dom/extend-expect'
import { MemoryRouter } from 'react-router-dom'

describe('Nav', () => {
  it('renders Nav correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Nav />
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
