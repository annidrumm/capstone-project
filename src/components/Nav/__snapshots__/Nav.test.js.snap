// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders Nav correctly 1`] = `
Array [
  <img
    alt="arrow left"
    className="GoBackArrow__GoBackStyled-m93ye8-0 kcjJZM"
    id="goBack"
    onClick={[Function]}
    src="back.svg"
    value="goBack"
  />,
  <button
    className="Nav__ButtonStyled-sc-8pthb9-0 jVkSCe"
  >
    <a
      href="/favourites"
      onClick={[Function]}
    >
      <svg
        fill="var(--darkpink)"
        stroke="var(--darkpink)"
      >
        heart.svg
      </svg>
    </a>
  </button>,
]
`;
